### PR TITLE
feat(router): switch to v2 routing store by default

### PR DESCRIPTION
**Pull request by keynslug**
Part of [EMQX-10713](https://emqx.atlassian.net/browse/EMQX-10713).

Followup to #11524.

### <samp>🤖 Generated by Copilot at f363fc3</samp>

Change the default routing storage schema from `v1` to `v2` for better performance and scalability. Update `emqx_schema.erl` to reflect this change.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] ~~Added property-based tests for code which performs user input validation~~
- [x] Changed lines covered in coverage report
- [x] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] ~~Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket~~
- [x] Schema changes are backward compatible

### CODE REVIEW

File: changes/ee/feat-11935.en.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Switch to the new `v2` routing store schema by default. New schema improves both subscription and routing performance, especially so for scenarios with concurrent subscriptions to topic filters sharing common wildcard prefixes, at the cost of slightly increased memory usage. This schema also eliminates the need for a separate index, thus inconsistencies in the routing state rarely encountered in previous versions should no longer be possible.
+
+Full non-rolling cluster upgrade or restart is required to switch to the new default, otherwise the former schema will still be in use.
```
Comment (zmstone) [inline changes/ee/feat-11935.en.md:3]:
```suggestion
If a cluster is rolling upgraded from older version, the cluster will continue to use `v` store until a full cluster (non-rolling) restart happens.
```

File: apps/emqx_utils/src/emqx_utils_stream.erl [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+        end
+    end.
+
+%%
+
+-spec next(stream(T)) -> next(T) | [].
+next(S) ->
+    S().
+
+-spec take(non_neg_integer(), stream(T)) -> {[T], stream(T)} | [T].
```
Comment (zmstone) [inline apps/emqx_utils/src/emqx_utils_stream.erl:87]: add doc ?

Comment (zmstone) [reply apps/emqx_utils/src/emqx_utils_stream.erl:87]: maybe another flavor of `consume` is better than `take`

File: apps/emqx_management/src/emqx_mgmt_api_topics.erl [hunk 3]
```diff
@@ -110,23 +107,15 @@ topic(get, #{bindings := Bindings}) ->
...
-            ?ROUTE_TAB,
-            Params,
-            ?TOPICS_QUERY_SCHEMA,
-            fun ?MODULE:qs2ms/2,
-            fun ?MODULE:format/1
-        )
-    of
-        {error, page_limit_invalid} ->
-            {400, #{code => <<"INVALID_PARAMETER">>, message => <<"page_limit_invalid">>}};
-        {error, Node, Error} ->
```
Comment (zmstone) [inline apps/emqx_management/src/emqx_mgmt_api_topics.erl:125]: there is no need to handle this error ?

File: apps/emqx_utils/src/emqx_utils_stream.erl (hunk 2 above)
Comment (keynslug) [reply apps/emqx_utils/src/emqx_utils_stream.erl]: ✅

File: apps/emqx/src/emqx_router.erl [hunk 4]
```diff
@@ -436,11 +472,52 @@ get_dest_node({_, Node}) ->
...
+mk_route_stream(filter, {MTopic, MDest}) ->
+    emqx_utils_stream:map(
+        fun routeidx_to_route/1,
+        emqx_utils_stream:ets(
+            fun
+                (undefined) ->
+                    MatchSpec = #routeidx{entry = emqx_trie_search:make_pat(MTopic, MDest)},
+                    ets:match_object(?ROUTE_TAB_FILTERS, MatchSpec, 1);
+                (Cont) ->
+                    ets:match_object(Cont)
```
Comment (zmstone) [inline apps/emqx/src/emqx_router.erl:506]: maybe there is no need for the map stream if we do lists:map/2 for the results before return ?

File: apps/emqx_utils/src/emqx_utils_stream.erl [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+%% @doc Make a stream out of an ETS table, where the ETS table is scanned through in chunks,
+%% with the given continuation function. The function is assumed to return a result of a call to:
+%% * `ets:select/1` / `ets:select/3`
+%% * `ets:match/1` / `ets:match/3`
+%% * `ets:match_object/1` / `ets:match_object/3`
+-spec ets(fun((Cont) -> select_result(Record, Cont))) -> stream(Record).
+ets(ContF) ->
+    ets(undefined, ContF).
+
+ets(Cont, ContF) ->
```
Comment (zmstone) [inline apps/emqx_utils/src/emqx_utils_stream.erl:136]: nit, maybe no need for two case clauses if we do it like this:
```suggestion
ets('$end_of_table', _ContF) ->
    empty();
ets(Cont, ContF) ->
```

Comment (zmstone) [reply apps/emqx_utils/src/emqx_utils_stream.erl:136]: sorry, the second clause is a special case for empty tables.

**Review by zmstone — APPROVED** [review]

File: apps/emqx/src/emqx_router.erl (hunk 4 above)
Comment (keynslug) [reply apps/emqx/src/emqx_router.erl:506]: You mean as part of `ContF`? We can, right. Might be a bit messy due to the need to handle 2 different possible shapes of a return value.

**Review by zmstone — APPROVED** [review]

**Pull request merged**